logstash-core/spec/logstash/pipeline_spec.rb

```diff
@@ -762,9 +762,12 @@ class TestPipeline < LogStash::Pipeline
     end
     let(:dummyoutput) { ::LogStash::Outputs::DummyOutput.new({ "id" => dummy_output_id }) }
     let(:metric_store) { subject.metric.collector.snapshot_metric.metric_store }
+    let(:before_subject_hook) {}
     let(:pipeline_thread) do
       # subject has to be called for the first time outside the thread because it will create a race condition
       # with the subject.ready? call since subject is lazily initialized
+      # The hook is used to allow support for mocking,etc in tests
+      before_subject_hook
```

Comment:
This is an interesting workaround. It doesn't feel like the right way to do this, though I am not able to precisely articulate exactly why I feel this way.